packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -380,10 +380,34 @@ Future<void> buildVariablesTree(
       }
     }
   }
-  if (instanceRef != null && serviceManager.service != null) {
+
+  if (variable.childCount > Variable.MAX_CHILDREN_IN_GROUPING) {
+    final numChildrenInGrouping = variable.childCount >=
+            Variable.MAX_CHILDREN_IN_GROUPING *
+                Variable.MAX_CHILDREN_IN_GROUPING
+        ? (roundToNearestPow10(variable.childCount) /
+                Variable.MAX_CHILDREN_IN_GROUPING)
+            .floor()
+        : Variable.MAX_CHILDREN_IN_GROUPING;
+
+    var index = 0;
+    while (index + numChildrenInGrouping < variable.childCount) {
+      variable.addChild(
+        Variable.grouping(variable.ref,
+            offset: index + (variable.offset ?? 0),
+            count: numChildrenInGrouping),
+      );
+      index += numChildrenInGrouping;
+    }
+    variable.addChild(
+      Variable.grouping(variable.ref,
+          offset: index + (variable.offset ?? 0),
```

Comment:
at this point, wouldn't index be > variable.count since `index += numChildrenInGrouping` just occurred?

Reply:
Ah, no because of the while stopping condition. But I've refactored to make this cleaner